tests/CommandsTest.php

```diff
@@ -179,6 +179,17 @@
         ->expectsOutput('Tenant: ' . $tenantId2);
 });
 
+test('run command works when sub command asks question and accepts argument', function () {
+    $id = Tenant::create()->getTenantKey();
+
+    Artisan::call('tenants:migrate', ['--tenants' => [$id]]);
+
+    pest()->artisan("tenants:run --tenants=$id 'age:ask Abrar' ")
+        ->expectsQuestion('What is your age?', 22)
+        ->expectsOutput("Tenant: $id")
+        ->expectsOutput("Abrar's age is 22.");
```

Comment:
Do these tests verify that the tenant context is *actually* used for the command?

I guess some of the other tests in `CommandsTest` do?

Reply:
> Do these tests verify that the tenant context is actually used for the command?

Yes, they do. `->expectsOutput("Tenant: $id")` verifies that we are in tenant context.

Reply:
Ah, I see 👍🏻

Reply:
Hmm I looked at it now and I'm not sure if it asserts that we're *inside* the tenant's context?

`tenants:run` simply prints the id of the tenant. But we're not asserting that the code you're using is actually *using* the tenant context.

The reason why I want to test this in detail is that you're instantiating the `Kernel` in the implementation (which we'd ideally not do), so I'm not 100% sure if the tenant context is shared.

Reply:
I got the point. 

So now I've updated the tests and command too. TestCommand now creates a user. So in tests, I'm asserting that the user created in tenant context, not in central. I believe this ensures we are running commands in the tenant context.